drivers: pwm: pwm_stm32: reset timer using RCC before initialization

sknsean opened the pull request:
If a timer is left running on an stm32mp1, (most likely) on the next run the timer is stuck.
A simple timer reset before initialization fixes the issue.

sknsean committed:
drivers: pwm: pwm_stm32: reset timer using RCC before initialization

If a timer is left running on an stm32mp1, (most likely) on the next run
the timer is stuck.
A simple timer reset before initialization fixes the issue.

Signed-off-by: Sean Nyekjaer <sean@geanix.com>

henrikbrixandersen commented on drivers/pwm/pwm_stm32.c:
```diff
@@ -678,6 +681,9 @@ static int pwm_stm32_init(const struct device *dev)
 	}
 
+	/* Reset timer to default state using RCC */
+	reset_line_toggle_dt(&data->reset);
```
Wouldn't this reset _all_ PWM instances below the parent timer instance?

sknsean replied:
Yes...
If that's an issue here, guess it also would be an issue here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/counter/counter_ll_stm32_timer.c

FRASTM commented on drivers/pwm/pwm_stm32.c:
```diff
@@ -753,7 +759,10 @@ static void pwm_stm32_irq_config_func_##index(const struct device *dev)		\
-	static struct pwm_stm32_data pwm_stm32_data_##index;                   \
+	static struct pwm_stm32_data pwm_stm32_data_##index = {                \
+		.reset = RESET_DT_SPEC_GET(DT_INST_PARENT(index)),	       \
+	};								       \
```
Like in the `counter_ll_stm32_timer.c`, it could written adding a macro 
`#define PWM(idx)              DT_INST_PARENT(idx)`

FRASTM commented on drivers/pwm/pwm_stm32.c:
```diff
@@ -678,6 +681,9 @@ static int pwm_stm32_init(const struct device *dev)
 	}
 
+	/* Reset timer to default state using RCC */
+	reset_line_toggle_dt(&data->reset);
```
(void)reset_line_toggle_dt  ?

sknsean committed:
drivers: pwm: pwm_stm32: add macro for using DT_INST_PARENT()

Make device init more readable by adding a macro

Suggested-by: Francois Ramu <francois.ramu@st.com>
Signed-off-by: Sean Nyekjaer <sean@geanix.com>
